pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -715,6 +728,15 @@ func (r *ReconcileClusterPool) createCluster(
 		}
 		poolRef := poolReference(clp)
 		cd.Spec.ClusterPoolRef = &poolRef
+		if clp.Spec.Inventory != nil {
+			if err := r.getCustomizedInstallConfigSecretRef(cd, customization, logger); err != nil {
+
+			}
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to customize install config")
+			}
+			cd.Spec.ClusterPoolRef.ClusterDeploymentCustomizationRef = customization
+		}
```

Comment:
this code will never be executed since we process only for obj of type `hivev1.ClusterDeployment`, otherwise skip. We need to add logic for detecting type `InstallConfig`